matchbox environment start fail on x86-64 target, while ok on x86 target. Root
cause is libmatchbox use "0"(int) as termination indicator when calling
XftFontOpen, which in turn called FcPatternVapBuild(in fontconfig). It try to
get the "0" as char* and fetch wrong value, as int and char* has different size
on x86-64. This patch forces a NULL pointer as terminator to fix it.

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>

Upstream-Status: Accepted

Index: libmatchbox-1.9/libmb/mbexp.c
===================================================================
--- libmatchbox-1.9.orig/libmb/mbexp.c	2010-08-28 06:33:25.000000000 +0800
+++ libmatchbox-1.9/libmb/mbexp.c	2010-08-28 06:30:05.000000000 +0800
@@ -348,7 +348,7 @@
 			    XFT_SIZE, XftTypeDouble   , (double)font->pt_size,
 			    XFT_WEIGHT, XftTypeInteger, weight,
 			    XFT_SLANT, XftTypeInteger , slant,
-			    0);
+			    NULL);
 
   if (font->font != NULL ) result = 2;
 
